dataprofiler/data_readers/csv_data.py

```diff
@@ -231,7 +235,64 @@ def _guess_header_row(data_as_str, suggested_delimiter=None,
         # If change in statistics never occurs, return no header
         if not change_flag:
             row_classic_header_ends = None
+
+
+        # Attempt to resolve case where only strings in every cell
+        if only_string_flag:
+            col_stats = {}
+            rows = data_as_str.split('\n')
+            for i in range(0, len(rows)):
+                cells = list(csv.reader([rows[i]],
+                                        delimiter=delimiter,
+                                        quotechar='"'))[0]
+                for j in range(0, len(cells)):
+
+                    # Determine number of words in cell
+                    word_count = 0
+                    if len(cells[j].strip()) > 0:
+                        words = cells[j].strip().split(' ')
+                        word_count = len(words)
                     
+                    # First row, set base
+                    if j not in col_stats:
+                        col_stats[j] = {"max":word_count, "min":word_count}
+
+                    # Identify min / max for a column
+                    if word_count > col_stats[j]['max']:
+                        col_stats[j]['max'] = word_count
+                    if word_count < col_stats[j]['min']:
+                        col_stats[j]['min'] = word_count
+
+                    # First index with value
+                    if 'first_index_with_value' not in col_stats[j] and word_count > 0:
+                        col_stats[j]['first_index_with_value'] = (i, word_count) 
```

Comment:
You never actually index into [1] for these entries to access the word count, is it still necessary to store it then?